docs/content/preview/integrations/spring-framework/sd-jpa.md

```diff
@@ -0,0 +1,249 @@
+
+  long count();                        
+
+  void delete(T entity);               
+
+  boolean existsById(ID primaryKey);   
+
+  // … more functionality omitted.
+}
+```
+
+Spring boot application can be configured to enable these repositories by using the following Java configuration class.
+
+```java
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
+
+@EnableJpaRepositories
+class Config { … }
+```
+
+#### Insert Data
+
+To write data in to YugabyteDB, implement the repository for each domain object.
+
+For example
+
+```java
+import org.springframework.data.jpa.repository.JpaRepository;
+import com.yugabyte.springdemo.model.User;
+
+public interface UserRepository extends JpaRepository<User, Long> {
+
+}
+```
+
+This repository can then be used for inserting data in to `Users` table.
+
+```java
+
+@Autowired
+private UserRepository userRepository;
+
+...
+
+@PostMapping("/users")
+public User createUser(@Valid @RequestBody User user) {
+    return userRepository.save(user);
+}
+```
+
+#### Query Data
+
+To write data in to YugabyteDB, use the out of the box query methods exposed by the repositories.
+
+For Example
+
+```java
+@Autowired
+private UserRepository userRepository;
+
+@PutMapping("/users/{userId}")
+public User updateUser(@PathVariable Long userId,
+                                @Valid @RequestBody User userRequest) {
+    return userRepository.findById(userId)
+            .map(user -> {
+                user.setFirstName(userRequest.getFirstName());
+                user.setLastName(userRequest.getLastName());
+                user.setEmail(userRequest.getEmail());
+                return userRepository.save(user);
+            }).orElseThrow(() -> new ResourceNotFoundException("User not found with id " + userId));
+}
+```
+
+Spring Data JPA repositories provides [named queries](https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#jpa.query-methods.at-query) for adding custom queries to repositories and to update the behavior of in built queries by using `@Query` annotation in the repository implementation.
+
+For Example
+
+```java
+import org.springframework.data.jpa.repository.JpaRepository;
+import com.yugabyte.springdemo.model.User;
+
+public interface UserRepository extends JpaRepository<User, Long> {
+
+  @Query("select u from User u where u.emailAddress = ?1")
+  User findByEmailAddress(String emailAddress);
+}
+```
+
+### Transaction and Isolation Levels
+
+YugabyteDB supports transactions for inserting and querying data from the tables. YugabyteDB supports different [isolation levels](../../../../architecture/transactions/isolation-levels/) for maintaining strong consistency for concurrent data access.
+
+By default, CRUD methods on JPA repository are transactional. For read operations, the transaction configuration `readOnly` flag is set to true. All others are configured with a plain `@Transactional` so that default transaction configuration applies.
```

Comment:
```suggestion
By default, CRUD methods on a JPA repository are transactional. For read operations, the transaction configuration `readOnly` flag is set to true. All others are configured with a plain `@Transactional` so that the default transaction configuration applies.
```